### PR TITLE
Massive knob changes

### PR DESCRIPTION
**Pull request by LunaeSomnia**

### CODE REVIEW

File: crates/vizia_core/src/views/knob.rs [hunk 1]
```diff
@@ -10,143 +10,252 @@ static DEFAULT_MODIFIER_SCALAR: f32 = 0.04;
...
+#[derive(Lens)]
+pub struct Knob<L: Lens<Target = D>, D: Copy + Clone + Into<f32> + 'static> {
+    lens: L,
     is_dragging: bool,
     prev_drag_y: f32,
-    continuous_normal: f32,
+    cursor: D,
+    default_normal: D,
+    min_value: D,
+    max_value: D,
```
Comment (Fredemus) [inline crates/vizia_core/src/views/knob.rs:28]: I don't love the idea of knobs having a mix and min value. In my experience it's easier to have that applied to whatever the knob is changing in an event, especially if the curve between min and max isn't linear

**Review by Fredemus — COMMENTED** [review]
I like all of these changes, except for the knobs having min and max values. It makes them harder to use in the plugin case at least

File: crates/vizia_core/src/views/knob.rs (hunk 1 above)
Comment (LunaeSomnia) [reply crates/vizia_core/src/views/knob.rs:28]: Makes total sense. I haven't even considered non-linear while re-implementing the knob 😅 

I think a possible solution to this is not exposing the `min_value` and `max_value` from the constructor and have them fixed at `0.0` and `1.0` respectively for the continuous knob. 

How does that sound to you?

Comment (Fredemus) [reply crates/vizia_core/src/views/knob.rs:28]: Yeah that sounds like a fine solution to me

**Review by geom3trik — APPROVED** [review]

**Comment by geom3trik** [comment]
Just FYI I'm holding out on merging this until I've finished the accesskit PR because in that PR I had to make some changes to sliders to make them accessible and I'd like to do the same with the knob.

**Pull request closed without merging**